privval/src/signer.rs

```diff
@@ -0,0 +1,74 @@
+//! Signing-related interface and the sample software-only implementation.
+
+use async_signature::AsyncSigner;
+use ed25519_consensus::SigningKey;
+use rand_core::{CryptoRng, RngCore};
+use tendermint::{account, PrivateKey, PublicKey, Signature};
+
+/// The trait for different signing backends
+/// (HSMs, TEEs, software-only, remote services etc.)
+#[tonic::async_trait]
+pub trait SignerProvider: AsyncSigner<Signature> {
+    async fn load_pubkey(&self) -> Result<PublicKey, async_signature::Error>;
+}
```

Comment:
Have you considered a blanket impl similar to this one here?

https://docs.rs/cosmrs/latest/cosmrs/crypto/secp256k1/trait.EcdsaSigner.html#impl-EcdsaSigner

The nice thing about that is it would allow this trait to "just work" with types that impl the `Signer` trait, leveraging a blanket impl of `AsyncSigner` for `Signer`:

https://docs.rs/async-signature/latest/async_signature/trait.AsyncSigner.html#implementors